From 01eba10373292d10a5e0f97e589c9a95980ce14f Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario_limonciello@dell.com>
Date: Thu, 16 Jul 2015 11:59:17 -0500
Subject: [PATCH 2/3] Remove update-online and update-offline functions.

This is for focusing on just providing an 'install' functions
that will detect whether to install online or offline when
appropriate.
---
 man/fwupdmgr.sgml | 20 --------------------
 src/fu-util.c     | 47 -----------------------------------------------
 2 files changed, 67 deletions(-)

diff --git a/man/fwupdmgr.sgml b/man/fwupdmgr.sgml
index 63843b7..8667000 100644
--- a/man/fwupdmgr.sgml
+++ b/man/fwupdmgr.sgml
@@ -118,26 +118,6 @@
       </varlistentry>
       <varlistentry>
         <term>
-          <option>update-offline</option>
-          <parameter>guid</parameter>
-          <parameter>filename</parameter>
-        </term>
-        <listitem>
-          <para>Install the update the next time the computer is rebooted.</para>
-        </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
-          <option>update-online</option>
-          <parameter>guid</parameter>
-          <parameter>filename</parameter>
-        </term>
-        <listitem>
-          <para>Install the update now, if possible.</para>
-        </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
           <option>update-prepared</option>
         </term>
         <listitem>
diff --git a/src/fu-util.c b/src/fu-util.c
index c7cae47..d2f3f0a 100644
--- a/src/fu-util.c
+++ b/src/fu-util.c
@@ -463,23 +463,6 @@ fu_util_update (FuUtilPrivate *priv, const gchar *id, const gchar *filename,
 }
 
 /**
- * fu_util_update_online:
- **/
-static gboolean
-fu_util_update_online (FuUtilPrivate *priv, gchar **values, GError **error)
-{
-	if (g_strv_length (values) != 2) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "Invalid arguments: expected 'id' 'filename'");
-		return FALSE;
-	}
-	return fu_util_update (priv, values[0], values[1],
-			       priv->flags, error);
-}
-
-/**
  * fu_util_install:
  **/
 static gboolean
@@ -729,24 +712,6 @@ fu_util_update_prepared (FuUtilPrivate *priv, gchar **values, GError **error)
 }
 
 /**
- * fu_util_update_offline:
- **/
-static gboolean
-fu_util_update_offline (FuUtilPrivate *priv, gchar **values, GError **error)
-{
-	if (g_strv_length (values) != 2) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "Invalid arguments: expected 'id' 'filename'");
-		return FALSE;
-	}
-	return fu_util_update (priv, values[0], values[1],
-			       priv->flags | FU_PROVIDER_UPDATE_FLAG_OFFLINE,
-			       error);
-}
-
-/**
  * fu_util_clear_results:
  **/
 static gboolean
@@ -1319,18 +1284,6 @@ main (int argc, char *argv[])
 		     _("Get all devices that support firmware updates"),
 		     fu_util_get_devices);
 	fu_util_add (priv->cmd_array,
-		     "update-offline",
-		     NULL,
-		     /* TRANSLATORS: command description */
-		     _("Install the update the next time the computer is rebooted"),
-		     fu_util_update_offline);
-	fu_util_add (priv->cmd_array,
-		     "update-online",
-		     NULL,
-		     /* TRANSLATORS: command description */
-		     _("Install the update now"),
-		     fu_util_update_online);
-	fu_util_add (priv->cmd_array,
 		     "update-prepared",
 		     NULL,
 		     /* TRANSLATORS: command description */
-- 
1.9.1

